### PR TITLE
iam-role and policy pr changes ghaction

### PR DESCRIPTION
**Pull request by pwyborn**
Why:
[Create Github Action to identify PRs that contain IAM Role and Policy changes#3015](https://app.zenhub.com/workspaces/cloud-platform-team-5ccb0b8a81f66118c983c189/issues/ministryofjustice/cloud-platform/3015)

### CODE REVIEW

**Review by liban-hmpps — APPROVED** [review]
LGTM

File: iam-role-policy-changes-check/tests/parse-pr_test.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package tests
+
+import (
+	"approve-doc-review/check"
```
Comment (poornima-krishnasamy) [inline iam-role-policy-changes-check/tests/parse-pr_test.go:4]:
```suggestion
       "iam-role-policy-changes-check/check"
```

File: iam-role-policy-changes-check/check/pr.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// Check is used to perform bool checks on whether a PR
+// contain potential IAM Role and Policy changes/additions 
+// to alert a reviewer to the above content
+package check
```
Comment (poornima-krishnasamy) [inline iam-role-policy-changes-check/check/pr.go:4]: Can you change the package name to something specific to the iam. This repo already have a package check under "approve-doc-preview"

File: iam-role-policy-changes-check/go,mod [hunk 3]
```diff
@@ -0,0 +1,8 @@
+module policy-changes
```
Comment (poornima-krishnasamy) [inline iam-role-policy-changes-check/go,mod:1]: This file is not needed

File: iam-role-policy-changes-check/go.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+module policy-changes
```
Comment (poornima-krishnasamy) [inline iam-role-policy-changes-check/go.txt:1]: This file is not needed

**Review by poornima-krishnasamy — COMMENTED** [review]
@pwyborn nice work on the policy check package. 
I have added few comments. 
Can you please also remove the `main` binary file which is not needed.

File: iam-role-policy-changes-check/tests/parse-pr_test.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package tests
+
+import (
+	"iam-role-policy-changes-check/identify-iam"
```
Comment (poornima-krishnasamy) [inline iam-role-policy-changes-check/tests/parse-pr_test.go:4]:
```suggestion
	"iam-role-policy-changes-check/identifyiam"
```

**Review by poornima-krishnasamy — APPROVED** [review]
Looks good

**Pull request merged**